test/functional/test_framework/evm_contracts.py

```diff
@@ -0,0 +1,42 @@
+import os
+from typing import List, Dict
+
+from solcx import compile_standard
+
+
+class EVMContract:
+    # Solidity compiler version
+    _solc_version = "0.8.10"
+    _path_prefix = "../contracts"
+
+    def __init__(self, file_name: str, contract_name: str):
```

Comment:
Could you apply the same construction conventions as the above for this as well and not do a error prone file read during construction? 

Let's init take content_str. Use a static helper `from_file` to construct it free to fallible methods in the ctor path.